build.gradle

```diff
@@ -81,54 +81,112 @@ task buildSconsArtifacts {
-            executable sconsExecutableFile.absolutePath
-            args "--directory=.", "platform=android", "target=template_debug", "arch=x86_64", "custom_api_file=thirdparty/godot_cpp_gdextension_api/extension_api.json", "build_profile=thirdparty/godot_cpp_build_profile/build_profile.json"
-        }
-        tasks.create(name: "buildGodotOpenXRVendorsAndroidX86_64Release", type: Exec) {
-            executable sconsExecutableFile.absolutePath
-            args "--directory=.", "platform=android", "target=template_release", "arch=x86_64", "custom_api_file=thirdparty/godot_cpp_gdextension_api/extension_api.json", "build_profile=thirdparty/godot_cpp_build_profile/build_profile.json"
-        }
+        createSconsTasks(sconsExecutableFile, false)
 
+        // Android.
         dependsOn 'buildGodotOpenXRVendorsAndroidArm64Debug'
         dependsOn 'buildGodotOpenXRVendorsAndroidArm64Release'
         dependsOn 'buildGodotOpenXRVendorsAndroidX86_64Debug'
         dependsOn 'buildGodotOpenXRVendorsAndroidX86_64Release'
 
-        // Build the GDExtension library for desktop.
-        tasks.create(name: "buildGodotOpenXRVendorsDesktopDebug", type: Exec) {
-            executable sconsExecutableFile.absolutePath
-            args "--directory=.", "target=template_debug", "custom_api_file=thirdparty/godot_cpp_gdextension_api/extension_api.json", "build_profile=thirdparty/godot_cpp_build_profile/build_profile.json"
-        }
-        tasks.create(name: "buildGodotOpenXRVendorsDesktopRelease", type: Exec) {
-            executable sconsExecutableFile.absolutePath
-            args "--directory=.", "target=template_release", "custom_api_file=thirdparty/godot_cpp_gdextension_api/extension_api.json", "build_profile=thirdparty/godot_cpp_build_profile/build_profile.json"
-        }
-
+        // Desktop.
         dependsOn 'buildGodotOpenXRVendorsDesktopDebug'
         dependsOn 'buildGodotOpenXRVendorsDesktopRelease'
     }
 }
 
+/**
+ * Scons clean for the project
+ */
+task cleanScons {
+    File sconsExecutableFile = getSconsExecutableFile()
+
+    // Using `doFirst` so the exception doesn't happen until this task actually runs.
+    doFirst {
+        if (sconsExecutableFile == null) {
+            throw new GradleException("Unable to find executable path for the 'scons' command.")
+        } else {
+            logger.debug("Found executable path for scons: ${sconsExecutableFile.absolutePath}")
+        }
+    }
+
+    if (sconsExecutableFile != null) {
+        createSconsTasks(sconsExecutableFile, true)
+
+        // Android.
+        dependsOn 'cleanGodotOpenXRVendorsAndroidArm64Debug'
+        dependsOn 'cleanGodotOpenXRVendorsAndroidArm64Release'
+        dependsOn 'cleanGodotOpenXRVendorsAndroidX86_64Debug'
+        dependsOn 'cleanGodotOpenXRVendorsAndroidX86_64Release'
+
+        // Desktop.
+        dependsOn 'cleanGodotOpenXRVendorsDesktopDebug'
+        dependsOn 'cleanGodotOpenXRVendorsDesktopRelease'
+    }
```

Comment:
Can the rest of the logic be moved within `createSconsTasks` as well? Seems like everything left out is identical in logic as well when taking into account the `build` / `clean` prefix switch.

Reply:
Not with my level of Gradle knowledge :-)

We need to have the exception stuff in a `doFirst { ... }` so it only happens when the task actually runs, and you can't put a `doFirst` in a function, it's just a task thing. Similar for the `dependsOn` - I don't know how to do that in a function rather than a task.